pkg/authentication/authenticator/authenticator.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Tencent is pleased to support the open source community by making TKEStack
+ * available.
+ *
+ * Copyright (C) 2012-2023 Tencent. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package authenticator
+
+import (
+	"sync"
+
+	"github.com/gin-gonic/gin"
+
+	"tkestack.io/kstone/pkg/authentication"
+	"tkestack.io/kstone/pkg/controllers/util"
+)
+
+const (
+	ProviderName = "generic"
+)
+
+var (
+	once     sync.Once
+	instance *Authenticator
+	AuthType = ""
```

Comment:
It is more appropriate to get global configuration from the config package here.